mempool/src/tests/integration_tests.rs

```diff
@@ -150,6 +154,172 @@ async fn vfn_middle_man_test() {
         .await;
 }
 
+/// Tests when a node skips an ack
+#[tokio::test]
+async fn test_skip_ack_rebroadcast() {
+    let all_txns = test_transactions(0, 2);
```

Comment:
nits: 
- I'd maybe reconsider refactoring some of the types around these transactions, slices and references. It would be cool if we didn't have to `as_slice()` and `&[1..2]`, etc. in every test. Even if the tests simple use `clone()` on these things, I don't think we care that much about performance.
- Also, it seems like transaction index 2 is unused in this test?

Reply:
Yeah, could probably clean that up.

At least the &[1..2] makes it clear which txn it is.